pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -164,26 +164,39 @@ func (r *ReconcileZookeeperCluster) Reconcile(request reconcile.Request) (reconc
 	return reconcile.Result{RequeueAfter: ReconcileTime}, nil
 }
 
-func zookeeperClusterFresherThanSts(instance *zookeeperv1beta1.ZookeeperCluster, sts *appsv1.StatefulSet) bool {
-	labeledRV, ok := sts.Labels["owner-rv"]
-	if !ok {
-		log.Info("Fail to find owner-rv in statefulset's labels; skip checking resource version")
-		return true
-	}
-	largerRV, err := strconv.Atoi(instance.ResourceVersion)
-	if err != nil {
-		log.Info("Fail to convert %s to integer; skip checking resource version", instance.ResourceVersion)
-		return true
+// compareResourceVersion compare resoure versions for the supplied ZookeeperCluster and StatefulSet
+// resources
+// Returns:
+//  0 if versions are equal
+// -1 if ZookeeperCluster version is less than StatefulSet version
+//  1 if ZookeeperCluster version is greater than StatefulSet version
+func compareResourceVersion(zk *zookeeperv1beta1.ZookeeperCluster, sts *appsv1.StatefulSet) int {
+
+	zkResourceVersion, zkErr := strconv.Atoi(zk.ResourceVersion)
+	stsVersion, stsVersionFound := sts.Labels["owner-rv"]
+
+	if !stsVersionFound {
+		if zkErr != nil {
+			log.Info("Fail to parse ZookeeperCluster version. Cannot decide zookeeper StatefulSet version")
+			return 0
+		}
+		return 1
 	}
-	smallerRV, err := strconv.Atoi(labeledRV)
+	stsResourceVersion, err := strconv.Atoi(stsVersion)
 	if err != nil {
-		log.Info("Fail to convert %s to integer; skip checking resource version", labeledRV)
-		return true
+		log.Info("Fail to convert StatefulSet version %s to integer; setting it to ZookeeperCluster version", stsVersion)
```

Comment:
Can we move this log message after line no:191

Reply:
why?

Reply:
doing that the info log would be missed if zkErr != nil

Reply:
if we are returning `0`,  we are not setting sts version to zookeepercluster version(in case of cluster doesnt have valid resource version)

Reply:
not sure I follow - can you be more clear?

Reply:
ok - I see your point. updated